packages/@react-aria/overlays/src/useOverlay.ts

```diff
@@ -149,6 +149,8 @@ export function useOverlay(props: AriaOverlayProps, ref: RefObject<Element>): Ov
     if (e.target === e.currentTarget) {
       e.preventDefault();
     }
+
+    isDismissable && onClose();
```

Comment:
Trying to think of cases where this is not actually the behavior we want (and if so I'll move this to Spectrum Popover to isolate).

Reply:
For cases where there are multiple dismissible overlays (popovers or modals without/without underlays), we will still have onInteractOutside block the body/underlay click from propagating beyond the top most overlay and thus only the outermost one will be closed which is the old expected behavior. For the case of our submenus/contextual help dialog, each nested popover is actually considered as non-dismissible because [isNonModal](https://github.com/adobe/react-spectrum/pull/4988/files#r1309415482) and our body/underlay click doesn't get its propagation stopped, allowing this line of code to execute and closes the root menu